app/config/AppStartupJobs.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config
+
+import models.{ProcessReadyCalculationRequest, ProcessedBulkCalculationRequest}
+import org.apache.pekko.actor.ActorSystem
+import org.mongodb.scala.MongoCollection
+import org.mongodb.scala.model.Filters
+import play.api.{Configuration, Logging}
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.CollectionFactory
+
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ExecutionContext, Future}
+
+
+class Startup @Inject()(
+                         appStartupJobs: AppStartupJobs,
+                         actorSystem: ActorSystem
+                       )(implicit ec: ExecutionContext) {
+
+  actorSystem.scheduler.scheduleOnce(FiniteDuration(1, TimeUnit.MINUTES)) {
+    appStartupJobs.runEverythingOnStartUp()
+  }
+}
+
+class AppStartupJobsImpl @Inject()(val config: Configuration,
+                                    val mongo: MongoComponent
+                                  )(implicit val ec: ExecutionContext) extends AppStartupJobs
+
+trait AppStartupJobs extends Logging {
+
+  implicit val ec: ExecutionContext
+  val mongo: MongoComponent
+
+  val processReadyCalsReqCollection: MongoCollection[ProcessReadyCalculationRequest] =
+    CollectionFactory.collection(mongo.database, "bulk-calculation", ProcessReadyCalculationRequest.formats)
+
+  val processedBulkCalsReqCollection: MongoCollection[ProcessedBulkCalculationRequest] =
+    CollectionFactory.collection(mongo.database, "bulk-calculation", ProcessedBulkCalculationRequest.formats)
+
+  private def logParentsMissingCreatedAtAndChildren(): Future[Unit] = {
+    val filter = Filters.and(
+      Filters.eq("isParent", true),
+      Filters.exists("createdAt", false)
+    )
+
+    processedBulkCalsReqCollection.find(filter).toFuture().flatMap { parents =>
```

Comment:
Might be nice to add a projection here to just get the bulk IDs we need?
This query pulls all of the data from the database when we don't need to actually read anything else?

Reply:
Yeah, exactly this - we want to be really careful we don't write a query that may cause another heap memory exception.